admin/admin_page_registration.php

```diff
@@ -5,91 +5,88 @@
  * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
  * @version $Id: DrByte 2020 Jun 07 Modified in v1.5.7 $
  */
-require('includes/application_top.php');
+require 'includes/application_top.php';
 
 // initialise form values
 $page_key = $language_key = $main_page = $page_params = $menu_key = $checked = '';
 $sort_order = 0;
 
 // check if we are receiving form content and if so validate and process it
 if (isset($_POST) && !empty($_POST)) {
-  $error = FALSE;
+    $error = FALSE;
 
-  if (isset($_POST['page_key'])) {
-    $page_key = zen_db_prepare_input($_POST['page_key']);
-  }
-  if (empty($page_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_PAGE_KEY_NOT_ENTERED, 'error');
-  } elseif (zen_page_key_exists($page_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_PAGE_KEY_ALREADY_EXISTS, 'error');
-  }
+    $keys = [
+        'page_key' => [
+            'empty' => true,
+            'elseif' => 'zen_page_key_exists',
+        ],
+        'language_key' => [
+            'empty' => true,
+            'elseif' => 'defined',
+            'elseifNot' => true,
+        ],
+        'main_page' => [
+            'empty' => true,
+            'elseif' => 'defined',
+            'elseifNot' => true,
+        ],
+        'page_params',
+        'menu_key' => [
+            'empty' => true,
+        ],
+        'sort_order' => [
+            'valueType' => 'integer',
+        ],
+    ];
 
-  if (isset($_POST['language_key'])) {
-    $language_key = zen_db_prepare_input($_POST['language_key']);
-  }
-  if (empty($language_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_LANGUAGE_KEY_NOT_ENTERED, 'error');
-  } elseif (!defined($language_key)) {
-    $error = TRUE;
-    $messageStack->add(ERROR_LANGUAGE_KEY_HAS_NOT_BEEN_DEFINED, 'error');
-  }
+    foreach ($keys as $key => $value) {
+        if (isset($_POST[$key])) {
+            if (!empty($value['valueType'])) {
+                ${$key} = $db->bindVars(':val:', ':val:', $_POST[$key], $value['valueType']);
+            } else {
+                ${$key} = zen_db_prepare_input($_POST[$key]);
+            }
+        }
+        if (!empty($value['empty'])) {
+            if (empty(${$key})) {
+                $error = TRUE;
+                $messageStack->add(constant('ERROR_' . strtoupper($key) . '_NOT_ENTERED'), 'error');
+            } else if ((!empty($value['elseif']) && $value['elseif'](${$key})) ? empty($value['elseifNot']) /* result was true */ : !empty($value['elseif']) && !empty($value['elseifNot']) /*result was false */) {

```

Comment:
The following is gonna sound highly critical. I hope it doesn't come off snobbish. It's primarily meant to foster discussion to create an understanding of preferred style, as I know you've invited these discussions elsewhere.

This `foreach/if/if/else/if/if/elseif/elseifNot/if(elseif)` loop refactoring makes my head spin.
I "get" it.
It's clever. 
Unfortunately next time I look at this code I'll have to spend 30 minutes figuring out what it's doing, all over again.

I appreciate the refactor.
And still I prefer to avoid the "arrow" anti-pattern, eg: if/if/else/if/if/else-if/if.

I don't follow the purpose of `empty`. Everything's `empty`, other than sort_order. Why not simplify it by just skipping sort_order when going through the loop?

Reply:
*I* certainly get it and a reason I put it out there... Most of the refactoring that has been happening causes the code to look nothing like it used to look, but expected to still function as it did... A smidge of "history" on this one, I wanted to literally take it one for one and even use the `elseif` key of `!defined`; however, doesn't work that way... :( So then after quite a lot of thought I finally figured out a logic path that *could* work and trying to consider some realm of "expansion" or reuse... Not all code can be that way, but I wanted to try to start as strong that way as possible.

In most of the cases, the `empty` test is intended to determine if the array's `key` has a value against which to take action and to consider the strict php case(s)... I certainly don't doubt that there are some (a lot?) of readability or other improvements possible. in earnest I don't mind if this stays open for a really long time... It just seems like there is/was an improved way to accomplish this repetitive series of checks and not require every possible array key to be fully filled in.

*Note*: both `sort_order` **and** `page_params` don't use the `empty` check...  The key `empty` implies to test if the variable/key is empty and if it is to take one action, if not to then do/try the second test/check if it is to be performed...